### PR TITLE
Update NTO/PPC to ginkgo v2

### PR DESCRIPTION
**Pull request by marioferh**
Update NTO and PPC to ginkgo v2

### CODE REVIEW

**Comment by yanirq** [comment]
On ci/prow/e2e-no-cluster we have a real failure:
```
which: no ginkgo in (/opt/google/protobuf/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/go/bin)
Downloading ginkgo tool
Running Functional Tests: test/e2e/performanceprofile/functests-performance-profile-creator
hack/run-perf-profile-creator-functests.sh: line 26: ginkgo: command not found 
```

**Comment by marioferh** [comment]
> On ci/prow/e2e-no-cluster we have a real failure:
> 
> ```
> which: no ginkgo in (/opt/google/protobuf/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/go/bin)
> Downloading ginkgo tool
> Running Functional Tests: test/e2e/performanceprofile/functests-performance-profile-creator
> hack/run-perf-profile-creator-functests.sh: line 26: ginkgo: command not found 
> ```

Yes, the rebase ruins it. Let me check

File: hack/run-functests.sh [hunk 1]
```diff
@@ -6,9 +6,8 @@ LATENCY_TEST_RUN=${LATENCY_TEST_RUN:-"false"}
 which ginkgo
 if [ $? -ne 0 ]; then
 	echo "Downloading ginkgo tool"
-	# drop -mod=vendor flags, otherwise the installation will fail
```
Comment (yanirq) [inline hack/run-functests.sh:9]: Don't we need keep this comment and the not keep the drop mod ?

File: Makefile [hunk 2]
```diff
@@ -94,14 +94,14 @@ $(GOBINDATA_BIN):
 
 test-e2e:
 	for d in core basic reboots reboots/sno; do \
-	  KUBERNETES_CONFIG="$(KUBECONFIG)" $(GO) test -v -timeout 40m ./test/e2e/$$d -ginkgo.v -ginkgo.noColor -ginkgo.failFast || exit; \
+	  KUBERNETES_CONFIG="$(KUBECONFIG)" $(GO) test -v -timeout 40m ./test/e2e/$$d -ginkgo.v -ginkgo.no-color ginkgo.fail-fast || exit; \
```
Comment (yanirq) [inline Makefile:97]: fail-fast vs failFast is a change from v1 to v2?

File: hack/run-functests.sh [hunk 3]
```diff
@@ -6,9 +6,8 @@ LATENCY_TEST_RUN=${LATENCY_TEST_RUN:-"false"}
...
-	# because of the package can not be installed under the vendor directory
-    GOFLAGS='' go install github.com/onsi/ginkgo/ginkgo@v1.16.5
+    go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo@latest
```
Comment (yanirq) [inline hack/run-functests.sh:9]: please evaluate if -mod=mod needs to be brought back here.
also I would keep pinning ginkgo not to the latest but to the version we vendor (v2.5.1)

File: hack/run-functests.sh [hunk 4]
```diff
@@ -26,7 +25,7 @@ fi
...
 # -v: print out the text and location for each spec before running it and flush output to stdout in realtime
 # -r: run suites recursively
-# --failFast: ginkgo will stop the suite right after the first spec failure
-# --flakeAttempts: rerun the test if it fails
-# -requireSuite: fail if tests are not executed because of missing suite
-GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --failFast -skipPackage="5_latency_testing" --flakeAttempts=2 -requireSuite ${GINKGO_SUITS} -- -junitDir /tmp/artifacts
+# --fail-fast: ginkgo will stop the suite right after the first spec failure
+# --flake-attempts: rerun the test if it fails
+# --require-suite: fail if tests are not executed because of missing suite
+GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --fail-fast --skip-package="5_latency_testing" --flake-attempts=2 --require-suite ${GINKGO_SUITS} --junit-report=/tmp/artifacts/
```
Comment (yanirq) [inline hack/run-functests.sh:31]: no new line at the end of the file.

File: hack/run-latency-testing.sh [hunk 5]
```diff
@@ -1,11 +1,12 @@
 #!/bin/bash
 
-GINKGO_SUITS=${GINKGO_SUITS:-"./test/e2e/performanceprofile/functests/0_config ./test/e2e/performanceprofile/functests/5_latency_testing"}
+GINKGO_SUITS=${GINKGO_SUITS:-"test/e2e/performanceprofile/functests-performance-profile-creator"}
 
 which ginkgo
 if [ $? -ne 0 ]; then
 	echo "Downloading ginkgo tool"
-	go install github.com/onsi/ginkgo/ginkgo
+    go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo@latest
```
Comment (yanirq) [inline hack/run-latency-testing.sh:8]: pin to a specific version please as above (v2.5.1)

Comment (yanirq) [reply hack/run-latency-testing.sh:8]: consider if -mod=mod is needed

File: hack/run-latency-testing.sh [hunk 6]
```diff
@@ -1,11 +1,12 @@
 #!/bin/bash
 
-GINKGO_SUITS=${GINKGO_SUITS:-"./test/e2e/performanceprofile/functests/0_config ./test/e2e/performanceprofile/functests/5_latency_testing"}
+GINKGO_SUITS=${GINKGO_SUITS:-"test/e2e/performanceprofile/functests-performance-profile-creator"}
```
Comment (yanirq) [inline hack/run-latency-testing.sh:3]: why did the path change here from ../5_latency_testing to ../functests-performance-profile-creator ?

File: hack/run-latency-testing.sh [hunk 7]
```diff
@@ -18,8 +19,7 @@ fi
...
 # -r: run suites recursively
-# --failFast: ginkgo will stop the suite right after the first spec failure
-# --flakeAttempts: rerun the test if it fails
-# -requireSuite: fail if tests are not executed because of missing suite
-GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r -requireSuite ${GINKGO_SUITS} -- -junitDir /tmp/artifacts
-
+# --fail-fast: ginkgo will stop the suite right after the first spec failure
+# --flake-attempts: rerun the test if it fails
+# --require-suite: fail if tests are not executed because of missing suite
+GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --fail-fast --flake-attempts=2 --require-suite ${GINKGO_SUITS} --junit-report=/tmp/artifacts
```
Comment (yanirq) [inline hack/run-latency-testing.sh:25]: no new line at the end of the file

File: hack/run-perf-profile-creator-functests.sh [hunk 8]
```diff
@@ -4,10 +4,9 @@ GINKGO_SUITS=${GINKGO_SUITS:-"test/e2e/performanceprofile/functests-performance-
 
 which ginkgo
 if [ $? -ne 0 ]; then
-    echo "Downloading ginkgo tool"
-	# drop -mod=vendor flags, otherwise the installation will fail
-	# because of the package can not be installed under the vendor directory
-    GOFLAGS='' go install github.com/onsi/ginkgo/ginkgo@v1.16.5
+	echo "Downloading ginkgo tool"
+    go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo@latest
```
Comment (yanirq) [inline hack/run-perf-profile-creator-functests.sh:8]: same comments as above

File: hack/run-render-command-functests.sh [hunk 9]
```diff
@@ -20,7 +19,7 @@ fi
...
 # -v: print out the text and location for each spec before running it and flush output to stdout in realtime
 # -r: run suites recursively
-# --failFast: ginkgo will stop the suite right after the first spec failure
-# --flakeAttempts: rerun the test if it fails
-# -requireSuite: fail if tests are not executed because of missing suite
-GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --failFast --flakeAttempts=2 -requireSuite ${GINKGO_SUITS} -- -junitDir /tmp/artifacts
+# --fail-fast: ginkgo will stop the suite right after the first spec failure
+# --flake-attempts: rerun the test if it fails
+# --require-suite: fail if tests are not executed because of missing suite
+GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --fail-fast --flake-attempts=2 --require-suite ${GINKGO_SUITS} --junit-report=/tmp/artifacts
```
Comment (yanirq) [inline hack/run-render-command-functests.sh:25]: new line

File: hack/run-render-command-functests.sh [hunk 10]
```diff
@@ -4,10 +4,9 @@ GINKGO_SUITS=${GINKGO_SUITS:-"test/e2e/performanceprofile/functests-render-comma
 
 which ginkgo
 if [ $? -ne 0 ]; then
-    echo "Downloading ginkgo tool"
-	# drop -mod=vendor flags, otherwise the installation will fail
-	# because of the package can not be installed under the vendor directory
-    GOFLAGS='' go install github.com/onsi/ginkgo/ginkgo@v1.16.5
+	echo "Downloading ginkgo tool"
+    go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo@latest
+    ginkgo version
```
Comment (yanirq) [inline hack/run-render-command-functests.sh:9]: same comments as above

**Comment by yanirq** [comment]
I see a binary files added here :  *.test  files. 
If these are you own local test files please remove them from this PR.

File: test/e2e/testing_manifests/sno/kernel_parameter_add_rm-child.yaml [hunk 11]
```diff
@@ -0,0 +1,21 @@
+apiVersion: tuned.openshift.io/v1
```
Comment (yanirq) [inline test/e2e/testing_manifests/sno/kernel_parameter_add_rm-child.yaml:1]: all the files under `test/e2e/testing_manifests/sno/` are new and do not seem related to this PR, is this something added from @jmencak's commit ?  or just a result of a non updated rebase?

File: test/e2e/reboots/machine_config_labels.go [hunk 12]
```diff
@@ -0,0 +1,116 @@
+package e2e
```
Comment (yanirq) [inline test/e2e/reboots/machine_config_labels.go:1]: @jmencak  is this part of NTO's migration to v2?

File: test/e2e/performanceprofile/functests/utils/utils.go [hunk 13]
```diff
@@ -7,21 +7,17 @@ import (
...
 
-	. "github.com/onsi/ginkgo"
-
 	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
 )
 
 const defaultExecTimeout = 2 * time.Minute
 
-func BeforeAll(fn func()) {
+func BeforeEach(fn func()) {
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/utils/utils.go:15]: what is the reason for this change?

File: test/e2e/performanceprofile/functests/7_test_test/test_suite_test_test.go [hunk 14]
```diff
@@ -0,0 +1,40 @@
+package pao_mustgather
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/7_test_test/test_suite_test_test.go:1]: this new file is part of moving to v2?  do we need to align and add before suite functions ?  if so, do we apply them in our code?

**Comment by yanirq** [comment]
@marioferh  PR is quite large to review. I have addressed most of the changes in my comments.
I would advice to go again over all the files in this PR and check if all of them are relevant and are properly based on top of @jmencak's commit and on top of the branch HEAD.

File: hack/run-functests.sh (hunk 3 above)
Comment (marioferh) [reply hack/run-functests.sh:9]: with -mod=mod does not work in the ci, it is an error and ginkgo is not found.
Ok to the version

File: hack/run-latency-testing.sh (hunk 5 above)
Comment (marioferh) [reply hack/run-latency-testing.sh:8]: same

File: hack/run-latency-testing.sh (hunk 6 above)
Comment (marioferh) [reply hack/run-latency-testing.sh:3]: ouh copy paste  error :S

File: hack/run-render-command-functests.sh (hunk 9 above)
Comment (marioferh) [reply hack/run-render-command-functests.sh:25]: done

File: hack/run-render-command-functests.sh (hunk 10 above)
Comment (marioferh) [reply hack/run-render-command-functests.sh:9]: done

File: hack/run-functests.sh (hunk 3 above)
Comment (marioferh) [reply hack/run-functests.sh:9]: > I see a binary files added here : *.test files. What are these files? If these are you own local test files please remove them from this PR.

Sorry trash of running test

File: hack/run-functests.sh (hunk 4 above)
Comment (marioferh) [reply hack/run-functests.sh:31]: done

File: test/e2e/performanceprofile/functests/utils/utils.go (hunk 13 above)
Comment (marioferh) [reply test/e2e/performanceprofile/functests/utils/utils.go:15]: Not sure.
Fix it

File: test/e2e/performanceprofile/functests/7_test_test/test.go [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+
+var mustGatherFailed bool
+var mustGatherContentDir string
+var mustGatherTarDir string
+var mustGatherTgz string
+
+var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", func() {
+	Context("PAO Mustgather Tests", func() {
+		It("Verify Generic cluster resource definitions are captured", func() {
+			fmt.Printf("FUUUUUUUUUUUUUU")
```
[the remark this answers is not in the document]
Comment (marioferh) [reply test/e2e/performanceprofile/functests/7_test_test/test.go:17]: it seems as some leftover from other PR :S

File: test/e2e/performanceprofile/functests/7_test_test/test_suite_test_test.go (hunk 14 above)
Comment (marioferh) [reply test/e2e/performanceprofile/functests/7_test_test/test_suite_test_test.go:1]: Remove

File: test/e2e/performanceprofile/functests/2_performance_update/test_suite_performance_update_test.go [hunk 16]
```diff
@@ -42,10 +41,11 @@ var _ = AfterSuite(func() {
...
-	rr := []Reporter{}
-	if ginkgo_reporters.Polarion.Run {
-		rr = append(rr, &ginkgo_reporters.Polarion)
-	}
-	rr = append(rr, junit.NewJUnitReporter("performance_update"))
-	RunSpecsWithDefaultAndCustomReporters(t, "Performance Addon Operator Update e2e tests", rr)
+	RunSpecs(t, "Performance Addon Operator configuration")
 }
+
+var _ = ReportAfterSuite("e2e serial suite", func(r Report) {
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/2_performance_update/test_suite_performance_update_test.go:47]: what is this addition ?

Comment (marioferh) [reply test/e2e/performanceprofile/functests/2_performance_update/test_suite_performance_update_test.go:47]: its the new way to add reports in ginkgo v2

File: hack/run-functests.sh [hunk 17]
```diff
@@ -6,9 +6,8 @@ LATENCY_TEST_RUN=${LATENCY_TEST_RUN:-"false"}
 which ginkgo
 if [ $? -ne 0 ]; then
 	echo "Downloading ginkgo tool"
-	# drop -mod=vendor flags, otherwise the installation will fail
-	# because of the package can not be installed under the vendor directory
-    GOFLAGS='' go install github.com/onsi/ginkgo/ginkgo@v1.16.5
+    go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo@v2.5.1
```
Comment (yanirq) [inline hack/run-functests.sh:9]: is there a reason to bring back `-mod=mod` ?  in case there is no presence of ginkgo will the installation succeed?

File: hack/run-perf-profile-creator-functests.sh [hunk 18]
```diff
@@ -20,7 +19,7 @@ fi
...
 # -v: print out the text and location for each spec before running it and flush output to stdout in realtime
 # -r: run suites recursively
-# --failFast: ginkgo will stop the suite right after the first spec failure
-# --flakeAttempts: rerun the test if it fails
-# -requireSuite: fail if tests are not executed because of missing suite
-GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --failFast --flakeAttempts=2 -requireSuite ${GINKGO_SUITS} -- -junitDir /tmp/artifacts
+# --fail-fast: ginkgo will stop the suite right after the first spec failure
+# --flake-attempts: rerun the test if it fails
+# --require-suite: fail if tests are not executed because of missing suite
+GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --fail-fast --flake-attempts=2 --require-suite ${GINKGO_SUITS} --junit-report=/tmp/artifacts
```
Comment (yanirq) [inline hack/run-perf-profile-creator-functests.sh:25]: new line missing

File: hack/run-latency-testing.sh (hunk 7 above)
Comment (yanirq) [inline hack/run-latency-testing.sh:25]: new line still missing

**Comment by yanirq** [comment]
> I see a binary files added here : *.test files. What are these files? If these are you own local test files please remove them from this PR.

@marioferh  I still see the presence of these file.   Are this needed ?

**Comment by yanirq** [comment]
Is this PR still considered as WIP?

File: test/e2e/performanceprofile/functests/2_performance_update/unit_report_performance_update.xml [hunk 19]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/2_performance_update/unit_report_performance_update.xml:1]: is this file suppose to be committed or is it a generate file that we should not add here?
In addition, if this is a generated file it would be advised to change its output path

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 20]
```diff
@@ -422,23 +421,23 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
-		//fetch the latest profile
-		profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
-		Expect(err).ToNot(HaveOccurred())
-		var oldMcpSelector, oldNodeSelector map[string]string
-		var removeLabels func(map[string]string, *corev1.Node) error
-
 		//fetch existing MCP Selector if exists in profile
 		BeforeEach(func() {
+			//fetch the latest profile
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:438]: Is this addition related to the PR ?

File: test/e2e/performanceprofile/functests/6_mustgather_testing/unit_report_must-gather.xml [hunk 21]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/unit_report_must-gather.xml:1]: same comments as above

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go (hunk 20 above)
Comment (marioferh) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:438]: ginkgo v2 is failing if there is a expect in the context outisde of beforeall or it, move it here in order to fix it.

File: test/e2e/performanceprofile/functests/6_mustgather_testing/unit_report_must-gather.xml (hunk 21 above)
Comment (marioferh) [reply test/e2e/performanceprofile/functests/6_mustgather_testing/unit_report_must-gather.xml:1]: right

**Comment by marioferh** [comment]
/retest-required 

Failing in test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:648 No changes there in this PR. Retest

**Comment by yanirq** [comment]
There seems to be a real error in e2e-gcp : 
` Setup Node not in Ordered Container
BeforeAll(func() {
/go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/performance.go:58
  [BeforeAll] setup nodes must appear inside an Ordered container.  They cannot
  be nested within other containers, even containers in an ordered container. `

File: test/e2e/reboots/operator_test.go [hunk 22]
```diff
@@ -19,7 +19,7 @@ var (
 func TestNodeTuningOperator(t *testing.T) {
 	gomega.RegisterFailHandler(ginkgo.Fail)
 
-	ginkgo.By("getting cluster ControlPlaneTopology")
+	util.Logf("getting cluster ControlPlaneTopology")
```
Comment (fromanirh) [inline test/e2e/reboots/operator_test.go:22]: why moving to `util.Logf`?

File: test/e2e/reboots/sno/operator_test.go [hunk 23]
```diff
@@ -19,7 +19,7 @@ var (
 func TestNodeTuningOperator(t *testing.T) {
 	gomega.RegisterFailHandler(ginkgo.Fail)
 
-	ginkgo.By("getting cluster ControlPlaneTopology")
+	util.Logf("getting cluster ControlPlaneTopology")
```
Comment (fromanirh) [inline test/e2e/reboots/sno/operator_test.go:22]: ditto, why moving to `util.Logf`?

File: test/e2e/reboots/operator_test.go (hunk 22 above)
Comment (marioferh) [reply test/e2e/reboots/operator_test.go:22]: this is a change from @jmencak but I think there a By cause a crash, change for util.Log you get the log and the test works. @jmencak could you correct my comment?

File: test/e2e/reboots/sno/operator_test.go (hunk 23 above)
Comment (marioferh) [reply test/e2e/reboots/sno/operator_test.go:22]: same

File: test/e2e/testing_manifests/sno/kernel_parameter_add_rm-child.yaml (hunk 11 above)
Comment (marioferh) [reply test/e2e/testing_manifests/sno/kernel_parameter_add_rm-child.yaml:1]: these files are from Jiri's one https://github.com/openshift/cluster-node-tuning-operator/pull/500/files

Not sure if they are necessary

File: local [hunk 24]
```diff
@@ -0,0 +1,27 @@
+Building Performance Profile Creator (PPC)
```
Comment (yanirq) [inline local:1]: please cleanup

Comment (marioferh) [reply local:1]: done

**Comment by marioferh** [comment]
Waiting for split ci: https://github.com/openshift/cluster-node-tuning-operator/pull/536 and https://github.com/openshift/release/pull/35029

File: test/e2e/performanceprofile/functests/1_performance/cpu_management.go [hunk 25]
```diff
@@ -364,7 +363,7 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 		})
 	})
 
-	Describe("Verification that IRQ load balance can be disabled per POD", func() {
+	FDescribe("Verification that IRQ load balance can be disabled per POD", func() {
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/1_performance/cpu_management.go:366]: Please go over the code and remove all programmatic focus

**Comment by yanirq** [comment]
@marioferh  when the PR is in shape for reviews please remove WIP from title and squash the commits into one

**Comment by marioferh** [comment]
> @marioferh when the PR is in shape for reviews please remove WIP from title and squash the commits into one

We need to push these ones before is ready: https://github.com/openshift/cluster-node-tuning-operator/pull/536 and https://github.com/openshift/release/pull/35029

**Comment by marioferh** [comment]
/hold
Waiting for https://github.com/openshift/cluster-node-tuning-operator/pull/536 and https://github.com/openshift/release/pull/35029
2_performance_update is skipped https://github.com/openshift/cluster-node-tuning-operator/blob/fc5c2786a01119719a01ccc29bf4ee94617fb0bc/hack/run-functests.sh#L31 until new lane is merged

**Comment by fromanirh** [comment]
please remove binaries from this PR (e.g. https://github.com/openshift/cluster-node-tuning-operator/pull/517/commits/e5ec96ce2d43e9225090ea1cb3c8f7bab3f629f6#diff-2a2dca23356ef4b545792f66504afb32495fdf5bca4c63dab87251dd72011593)

We should never add binaries to git and each exception to this hard rule should be very carefully vetted.

File: test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go [hunk 26]
```diff
@@ -6,14 +6,13 @@ import (
 	"os/exec"
 	"testing"
 
-	"github.com/onsi/ginkgo"
-	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/ginkgo/v2"
```
Comment (ffromani) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go:10]: why twice?

File: test/e2e/reboots/operator_test.go (hunk 22 above)
Comment (ffromani) [reply test/e2e/reboots/operator_test.go:22]: I'm fine moving to `util.Logf`, I'm just curious about the rationale

**Review by ffromani — COMMENTED** [review]
I think we're almost there

File: test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go (hunk 26 above)
Comment (ffromani) [reply test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go:10]: Actually we can fix this later, nevermind

**Comment by ffromani** [comment]
minor comments are not worth a resubmit. If the current version passes CI, we can and should merge it.

**Comment by yanirq** [comment]
There is a need to align with https://github.com/openshift/cluster-node-tuning-operator/pull/536  and update https://github.com/openshift/cluster-node-tuning-operator/blob/master/hack/run-update-testing.sh  (also check other files that were updated in this PR)

File: test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go (hunk 26 above)
Comment (marioferh) [reply test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go:10]: Done

File: test/e2e/reboots/operator_test.go (hunk 22 above)
Comment (marioferh) [reply test/e2e/reboots/operator_test.go:22]: As I remember after change to ginkgo v2 that crashes because By cannot be outside a ginkgo node

File: test/e2e/performanceprofile/functests/utils/utils.go [hunk 27]
```diff
@@ -7,14 +7,14 @@ import (
...
-
 	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+
+	. "github.com/onsi/ginkgo/v2"
 )
 
 const defaultExecTimeout = 2 * time.Minute
 
-func BeforeAll(fn func()) {
+func CustomBeforeAll(fn func()) {
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/utils/utils.go:17]: Why did we change the name here ? to override an original BefroeAll ?

Comment (marioferh) [reply test/e2e/performanceprofile/functests/utils/utils.go:17]: This must be fix in next PR to avoid issues with that custom BeforeAll func

**Comment by jmencak** [comment]
Thank you for the changes, @marioferh !  From the "core" NTO perspective this looks good.  The only change I'm missing from the [PR500](https://github.com/openshift/cluster-node-tuning-operator/pull/500/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L97) are the `s/failFast/fail-fast/` changes in the `Makefile`.  Can we include those, please?

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 28]
```diff
@@ -407,7 +405,7 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			}
 		})
 
-		It("Reverts back all profile configuration", func() {
+		AfterAll("Reverts back all profile configuration", func() {
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:408]: What do we gain from the change here?  
The current format will not work as it is:
```
 Unknown Decorator
AfterAll("Reverts back all profile configuration", func() {
/go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:408
  [AfterAll] node was passed an unknown decorator: '"Reverts back all profile
  configuration"' 
```

Comment (marioferh) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: Fix. An It cannot be used to return original config

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 29]
```diff
@@ -407,7 +405,7 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			}
 		})
 
-		It("Reverts back all profile configuration", func() {
+		AfterAll(func() {
```
Comment (yanirq) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:408]: Setup Node not in Ordered Container
AfterAll(func() {
/go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:408
  [AfterAll] setup nodes must appear inside an Ordered container.  They cannot
  be nested within other containers, even containers in an ordered container.

**Comment by marioferh** [comment]
ci/prow/e2e-gcp-pao-updating-profile

File: hack/run-update-testing.sh [hunk 30]
```diff
@@ -29,4 +28,4 @@ echo "Running Functional Tests: ${GINKGO_SUITS}"
 # --failFast: ginkgo will stop the suite right after the first spec failure
 # --flakeAttempts: rerun the test if it fails
 # -requireSuite: fail if tests are not executed because of missing suite
-GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --failFast --flakeAttempts=2 -requireSuite ${GINKGO_SUITS} -- -junitDir /tmp/artifacts
+GOFLAGS=-mod=vendor ginkgo $NO_COLOR --v -r --fail-fast --flake-attempts=2 -require-suite ${GINKGO_SUITS} --junit-report=report.xml
```
Comment (yanirq) [inline hack/run-update-testing.sh:31]: Ginkgo V2 updates (https://onsi.github.io/ginkgo/MIGRATING_TO_V2) 
"Finally, the default timeout has been reduced from 24h down to 1h. Users with long-running tests may need to adjust the timeout in their CI scripts."

"In V2 the timeout now applies to the entire test suite run so that ginkgo -r -timeout=1h is now guaranteed to exit after (about) one hour."

@marioferh  Are we setting a default timeout somewhere else ?   if not then we should add the timeout explicitly 
(starting with 5h to give enough wiggle room)
